src/producer.rs

```diff
@@ -0,0 +1,84 @@
+use rust_arroyo::backends::kafka::config::KafkaConfig;
+use rust_arroyo::backends::kafka::producer::KafkaProducer;
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+
+use rust_arroyo::types::{Topic, TopicOrPartition};
+
+use crate::types::{
+    CheckResult,
+};
+use rust_arroyo::backends::{Producer, ProducerError};
+
+#[derive(Debug, thiserror::Error)]
+pub enum ExtractCodeError {
+    #[error(transparent)]
+    SerializationError(#[from] serde_json::Error),
+    #[error(transparent)]
+    ProducerError(#[from] ProducerError),
+}
+
+pub async fn produce_checker_result(
+    result: CheckResult,
+    topic: Topic,
+    config: KafkaConfig,
+) -> Result<(), ExtractCodeError> {
+    let producer = KafkaProducer::new(config.clone());
```

Comment:
Why is the clone needed here? doesn't look like we're using the config anywhere else?